File: II course/Programming Lab/Figures/Figure_Classes_1/OpenFigure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Figure_Classes_1
{
    public abstract class OpenFigure : IShape
    {
        public double square()
        {
            return 0;
        } 

        public abstract double length();
        public abstract IShape shift(Point2D a);
        public abstract IShape rot(double phi);
        public abstract IShape symAxis(int i);
        public abstract bool cross(IShape i);
    }
}

File: III course/Java programming/FiguresFX/src/main/java/com/figures/abstractclasses/OpenFigure.java
package com.figures.abstractclasses;

import com.figures.figures.points.Point2D;
import com.figures.interfaces.IShape;

public abstract class OpenFigure implements IShape {

    @Override
    public double square() {
        return 0;
    }

    public abstract double length();
    public abstract IShape shift(Point2D a);
    public abstract IShape rot(double phi);
    public abstract IShape symAxis(int i);
    public abstract boolean cross(IShape i);
}
